src/cargo/core/resolver/mod.rs

```diff
@@ -680,17 +688,6 @@ impl Context {
         let deps = parent.dependencies();
         let deps = deps.iter().filter(|d| d.is_transitive() || dev_deps);
 
-        // Second, ignoring dependencies that should not be compiled for this
-        // platform
-        let deps = deps.filter(|d| {
-            match *method {
-                Method::Required{target_platform: Some(ref platform), ..} => {
-                    d.is_active_for_platform(platform)
```

Comment:
You may actually be able to remove this `is_active_for_platform` method entirely.